### ISSUE TITLE
Have a VM to get started developing with

### ISSUE THREAD

**Issue by pedromaltez** [codebar/planner#242]
The project isn't that complex but would this be desirable? I've been building one out anyway for my own selfish purposes of learning more about provisioning, but I could clean it up and submit it if it's something that you want.

**Comment by despo** [comment]
@pedromaltez that would be great idea. A VM with PG configured & data would be great, I think these are the main issues to get people started.

If you are up for it I can provide a fresh export of the data this weekend.

**Comment by pedromaltez** [comment]
Oh, great. I was/(am?) having issues with the data and the fact that you need a mailchimp api key (which isn't mentioned anywhere, but `rake db:seed` fails without one!), which I don't know how to fix without setting a key. But it doesn't actually load any data from mailchimp because there is no data in my fresh account and no corresponding mailing list set up because my account is completely new and I don't know what the list is supposed to look like. Would we then seed it with your dbdump instead of the current db:seed script? Guess that would fix all the issues I've had.

**Comment by despo** [comment]
Mailchimp is not really required for development - it's called when subscribing and unsubscribing to different groups to update our mailing lists. I missed this because I was using my own account when initially developing to verify.
 
https://github.com/codebar/planner/blob/master/lib/services/mailing_list.rb#L9 - this should probably change to reflect that, I should add an issue.

**Comment by pedromaltez** [comment]
@ despo, so yes I'd like a dbdump if you can do that! Put it in a gist? Or somewhere that I can get to it. The VM is basically done. But I think I'll need to discuss some of the details with you. We can do that in the PR though. :)

**Comment by gcapizzi** [comment]
Hi folks! Any news on this? Setting up my dev environment right now and going through the same problems (Postgres, Mailchimp) :smile:

**Comment by pedromaltez** [comment]
Hey Giuseppe, I'll post the PR up within a couple of hours. But still no db dump, so hopefully we sort that out soon after I issue the PR.

**Comment by pedromaltez** [comment]
Scratch that, I'll do it 'tomorrow'. Got home pretty late and I'm tired, but it will be up. First thing I'll do once I wake up. :P

**Comment by gcapizzi** [comment]
No worries! Thanks a lot for working on this :)
We can take care of the dump later, seed data should be enough for development.

**Comment by pedromaltez** [comment]
@gcapizzi, no, thank you for nudging me about this. I'm just sorting out some niggles with the provisioner.

**Comment by pedromaltez** [comment]
@gcapizzi #244 fixes the Mailchimp problems.

**Comment by pedromaltez** [comment]
@despo if you can merge #244 it would simplify the instructions needed to get what will be the PR for this to work - no need to mention get a Mailchimp API key anymore.

**Comment by pedromaltez** [comment]
@gcapizzi sorry it took me a while. The provisioner is at #247.

**Comment by deniseyu** [comment]
@pedromaltez what's the status on this? looking forward to having a VM :-)

**Comment by alexanderkustov** [comment]
https://desktop.nanobox.io/

This thing has appeared on the horizon, might be worth having a small look and play around...

**Comment by zaccolley** [comment]
Could use docker too

**Comment by alexanderkustov** [comment]
@deniseyu , @zaccolley had an hour to play around with the concept, let me know what you think.

Seems pretty painless for a beginner to run those 3 commands.

Might require a further README.md setup rewrite.

**Comment by alexanderkustov** [comment]
Also we might want to deploy this app to docker hub and even make it availible for everyone with https://kitematic.com/

that would make anyone setting this up such a breeze
